Fails to build on 32-bit architectures

Regressed by 0eac415ba2d9. According to [POSIX.1-2017](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/time.h.html) `tv_nsec` is of `long` type.

```rust
$ rustc -vV
rustc 1.75.0 (82e1608df 2023-12-21) (built from a source tarball)
binary: rustc
commit-hash: 82e1608dfa6e0b5569232559e3d385fea5a93112
commit-date: 2023-12-21
host: i686-unknown-freebsd
release: 1.75.0
LLVM version: 17.0.6

$ cargo build
[...]
error[E0308]: mismatched types
   --> src/utils/clock.rs:122:21
    |
122 |             tv_sec: tp.as_secs() as std::os::raw::c_longlong,
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `i64`

error[E0308]: mismatched types
   --> src/utils/clock.rs:123:22
    |
123 |             tv_nsec: tp.subsec_nanos() as std::os::raw::c_longlong,
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `i32`, found `i64`

error[E0308]: mismatched types
   --> src/utils/clock.rs:149:27
    |
149 |             lhs.tv_nsec + NANOS_PER_SEC - rhs.tv_nsec
    |                           ^^^^^^^^^^^^^ expected `i32`, found `i64`

error[E0277]: cannot add `i64` to `i32`
   --> src/utils/clock.rs:149:25
    |
149 |             lhs.tv_nsec + NANOS_PER_SEC - rhs.tv_nsec
    |                         ^ no implementation for `i32 + i64`
    |
    = help: the trait `std::ops::Add<i64>` is not implemented for `i32`
    = help: the following other types implement trait `std::ops::Add<Rhs>`:
              <i32 as std::ops::Add>
              <i32 as std::ops::Add<&i32>>
              <&'a i32 as std::ops::Add<i32>>
              <&i32 as std::ops::Add<&i32>>

error[E0308]: mismatched types
   --> src/utils/clock.rs:153:31
    |
153 |         debug_assert!(nanos < NANOS_PER_SEC);
    |                       -----   ^^^^^^^^^^^^^ expected `i32`, found `i64`
    |                       |
    |                       expected because this is `i32`
    |
help: you can convert `nanos` from `i32` to `i64`, matching the type of `NANOS_PER_SEC`
    |
153 |         debug_assert!(i64::from(nanos) < NANOS_PER_SEC);
    |                       ++++++++++     +
```

--- a/src/utils/clock.rs
+++ b/src/utils/clock.rs
@@ -119,8 +119,8 @@ impl<Kind> Ord for Time<Kind> {
 impl<Kind: NonNegativeClockSource> From<Duration> for Time<Kind> {
     fn from(tp: Duration) -> Self {
         let tp = Timespec {
-            tv_sec: tp.as_secs() as std::os::raw::c_longlong,
-            tv_nsec: tp.subsec_nanos() as std::os::raw::c_longlong,
+            tv_sec: tp.as_secs() as rustix::time::Secs,
+            tv_nsec: tp.subsec_nanos() as rustix::time::Nsecs,
         };
         Time {
             tp,
@@ -138,7 +138,7 @@ impl<Kind> From<Timespec> for Time<Kind> {
     }
 }
 
-const NANOS_PER_SEC: std::os::raw::c_longlong = 1_000_000_000;
+const NANOS_PER_SEC: rustix::time::Nsecs = 1_000_000_000;
 
 fn saturating_sub_timespec(lhs: Timespec, rhs: Timespec) -> Option<Duration> {
     if let Some(mut secs) = lhs.tv_sec.checked_sub(rhs.tv_sec) {

